IPython/core/history.py

```diff
@@ -49,26 +43,8 @@ def __exit__(self, *args, **kwargs):
         pass
 
 
-@decorator
-def needs_sqlite(f, self, *a, **kw):
-    """Decorator: return an empty list in the absence of sqlite."""
-    if sqlite3 is None or not self.enabled:
-        return []
-    else:
-        return f(self, *a, **kw)
-
-
-if sqlite3 is not None:
-    DatabaseError = sqlite3.DatabaseError
-    OperationalError = sqlite3.OperationalError
-else:
-    @undoc
-    class DatabaseError(Exception):
-        "Dummy exception when sqlite could not be imported. Should never occur."
-    
-    @undoc
-    class OperationalError(Exception):
-        "Dummy exception when sqlite could not be imported. Should never occur."
+DatabaseError = sqlite3.DatabaseError
+OperationalError = sqlite3.OperationalError
```

Comment:
This create actual test failures;  The decorator has the following statement:

```
if sqlite3 is None or not self.enabled:
```

Note the `or not self.enabled`; So this decorator is not only about sqlite being available.